### ISSUE TITLE
Add speechd module

### ISSUE THREAD

**Issue by jtojnar** [NixOS/nixpkgs#39500]
This should allow us to add more modules like https://github.com/NixOS/nixpkgs/pull/32979 and set the default ones.

**Comment by davidak** [comment]
This would still be nice.

**Comment by Lord-Valen** [comment]
Still relevant.

**Comment by Lord-Valen** [comment]
Could be handy for #30760.

**Comment by davidak** [comment]
Feel free to take over my linked draft PR.

**Comment by Lord-Valen** [comment]
We should probably lay out a clear image of what we want this to do first. Right now the package seems to be doing [what I would ordinarily expect a module to be handling.](https://github.com/NixOS/nixpkgs/blob/9656e85a15a0fe67847ee8cdb99a20d8df499962/pkgs/development/libraries/speechd/default.nix#L16-L23) We **could** simply have a module which that starts the systemd service provided by the package, but I like to think we could do a little better than that. For example, we could be checking the config for certain values (pulse, alsa, pipewire pulse/alsa options come to mind) and determine *within the module* which flags should be passed (or settings set), instead of handing that burden to the user. If I set `pipewire.pulse.enable = true` it should be fairly obvious that I would want speechd to use pulse. And if for some reason I didn't, we could have `speechd.audioMethod` to override, not unlike how pinentry is handled in the gnupg module. 
`speechd.settings` is an obvious addition. The [syntax](https://htmlpreview.github.io/?https://github.com/brailcom/speechd/blob/master/doc/speech-dispatcher.html#Configuration-file-syntax) could be troublesome. If a dotconf format works out, dedicated modules for speech engines open up some possibilities: speechd supports [generic modules](https://htmlpreview.github.io/?https://github.com/brailcom/speechd/blob/master/doc/speech-dispatcher.html#Configuration-of-the-Generic-Output-Module), which we can use to extend support to some other engines such as coqui TTS and place these behind `speechd.*`. Beyond that, we could have `speechd.modules` where you could define custom modules.

**Comment by jtojnar** [comment]
> package seems to be doing [what I would ordinarily expect a module to be handling](https://github.com/NixOS/nixpkgs/blob/9656e85a15a0fe67847ee8cdb99a20d8df499962/pkgs/development/libraries/speechd/default.nix#L16-L23)

That is because the options affect closure size. But for the backends/modules, the dependencies get mostly pulled in through the config files so if we only substitute the paths in the NixOS module, we will be able to have narrower closure and enable module without rebuilding speechd. This was the reason I wanted this in the first place.

We could have the NixOS module just copy files from the package (similarly to what fwupd does) and then patch them to correct paths there. Most people probably will not need to customize the settings beyond what modules are installed anyway. And if they really need to, they can always use `environment.etc`.
